substrate/primitives/transaction-storage-proof/src/runtime_api.rs

```diff
@@ -0,0 +1,76 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Runtime API definition for the transaction storage proof processing.
+
+use sp_runtime::traits::NumberFor;
+
+sp_api::decl_runtime_apis! {
+	/// Runtime API trait for transaction storage support.
+	pub trait TransactionStorageApi {
+		/// Get the actual value of a storage period in blocks.
```

Comment:
Storage period is the time we keep the data alive? 

Maybe the entire naming could be improved?

Reply:
> > Storage period is the time we keep the data alive?
> > Maybe the entire naming could be improved?
> 
> Ok, ok, will think about something better, I will fix tomorrow

The best I could come up with is renaming StoragePeriod to RetentionPeriod, fixed by: https://github.com/paritytech/polkadot-sdk/pull/10656/commits/9f77401aafb14f1c4f0ee5b3d7dc3b0a6efec7b7


@bkchr  Is this ok, or do you have another suggestion?